src/equil/MultiPhase.cpp

```diff
@@ -343,7 +343,11 @@ void MultiPhase::setPhaseMoleFractions(const size_t n, const doublereal* const x
         init();
     }
     ThermoPhase* p = m_phase[n];
-    p->setState_TPX(m_temp, m_press, x);
+    if ( p->isPure() ) {
+        p->setState_TP(m_temp, m_press);
+    } else {
+        p->setState_TPX(m_temp, m_press, x);
+    }
```

Comment:
Ideally, this check wouldn't be necessary. I'd argue this is less readable than the existing version.

Reply:
It may be, but the MultiPhase solvers differentiate between pure and multi-species phases elsewhere. Suggest to keep as it is inconsequential.

Reply:
Reverting.